superset/db_engine_specs.py

```diff
@@ -675,18 +653,20 @@ def handle_cursor(cls, cursor, query, session):
             hive.ttypes.TOperationState.RUNNING_STATE,
         )
         polled = cursor.poll()
+        import logging
         while polled.operationState in unfinished_states:
-            query = session.query(type(query)).filter_by(id=query.id)
+            query = session.query(type(query)).filter_by(id=query.id).one()
             if query.status == QueryStatus.STOPPED:
                 cursor.cancel()
                 break
-
             resp = cursor.fetch_logs()
-            if resp and resp.log:
-                progress = cls.progress(resp.log)
+            if resp and len(resp) > 0:
```

Comment:
Just a "nulls" vs "things that define __len__" not begin the same .. so should `resp` return an object (that's not null, but empty) .. lots of folks forget the `__nonzero__` method is a thing.